src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/PooledStringDictionary.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.PooledObjects;
+
+internal sealed partial class PooledStringDictionary<V>
+{
+    private static ObjectPool<PooledDictionary<string, V>>? s_poolCaseInsensitiveInstance;
+
+    public static PooledDisposer<PooledDictionary<string, V>> GetInstance(bool caseSensitive, out PooledDictionary<string, V> instance)
+    {
+        if (caseSensitive)
+            return PooledDictionary<string, V>.GetInstance(out instance);
+
+        s_poolCaseInsensitiveInstance ??= PooledDictionary<string, V>.CreatePool(StringComparer.OrdinalIgnoreCase);
```

Comment:
just make it a static readonly and always initialize.  it's not going to matter.

Reply:
Per our conversation, I'm going to keep it as such as if it were the other way, a pool would be created for every value type of a string dictionary and that idea bothered me.

Reply:
I think you could do something like this if you prefer. https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/SpecializedSymbolCollections.cs,e96e13892c141f98,references

Reply:
@alrz -- I'm not clear on how that would help me avoid allocating this static pool for every type V that this is called for. I really only want to create the case insensitive pool for V if this method is called indicating a desire to use a case insensitive comparer.

Reply:
The initializer won't run if you don't reach the field access. Perhaps something like: ([sharplab](https://sharplab.io/#v2:CYLg1APgAgDABFAjAOgCIEsCGBzAdgewGcAXdAY0IG4BYAKDvV2IFMAnXTAGwUQDYEATHADKAB2ZksndAC9mwYcVaNsAYXydOE0vlyE6AbzpwTCAMw9+ABXwb5ACUyEAFsObEAPEhgA+OAHF3GzsFJRVHFzdiAEk9YkxcMmYACgAjW24yJ2Y3PXRSADdmAEpjUyNaUyq4dAAzOGSswhzmPMKShAB2OGCtUOVcbAjXdy9EX2RAmLiEpOTimkrqkwLMVhqZxOY4AF4ejPlFAaGnEeJkQgB9UQzYklnmZABBTXwslnnF5ZNUZlSAV2wz0IzVYxGSjHuW2Q6n+TF2exgCzKyyg3Uh8S2X1MAF86Ci4KJlKsWJZBPsQkdwqcogSKt8Nix2FwyaxmJhgLpOABPOAAeVSACttL0PL0HDTRt4fH4rjcNHdMUldhS+sMomMJqo2ZgWL1klTBuoALaiNZsZB81jARhcaJ4fBs1TZZFLEx4+hu8xk8XADBkHQcVjczUAGjgADU/FNfYbsP7A2tuYqHh4o2kMnAmi02ugiqUvfTvnUGtncoR8nmOmjVfIE+hdEmw5GfJN3Cmtp8CVVVusMQ8VbGwoN643g2nW3Lbpsks9Xu8Uq6Gb8AUCniC2ODIQ8Yfg4cQEXAkdjUeiZ8wT3APQSiXnddskPwoEIh8dR0GQ1G6d3TIwmRxuEfOAdU5XAeX5IURQyMUDj9chE3HbxwyjWVrmnKFlT2X13ybJCWxhHU9QyA1hzUfBTXNVhLWtW1OHtAgnRdS8PRxIA==))

Reply:
@alrz -- These were really great suggestions and led me to look into this a bit more at whcih point I found that Roslyn already has existing pools for these structures. I'm still learning to *extensively* look around Roslyn for existing solutions before implementing something myself.